localstack-core/localstack/services/stepfunctions/asl/component/state/state_execution/state_task/service/state_task_service_api_gateway.py

```diff
@@ -174,6 +174,11 @@ def _headers_of(parameters: TaskParameters) -> Optional[dict]:
                 for forbidden_prefix in StateTaskServiceApiGateway._FORBIDDEN_HTTP_HEADERS_PREFIX:
                     if key.startswith(forbidden_prefix):
                         raise ValueError(f"The 'Headers' field contains unsupported values: {key}")
+
+                value = headers.get(key)
+                if isinstance(value, list):
+                    headers[key] = ",".join(value)
```

Comment:
Thanks for applying the changes! We can probably wrap this string body between brackets ("h1,h2" -> "[h1,h2]"). It's interesting how the test cases did not highlight this formatting change.

Reply:
That seems to be breaking other tests 😅
Will check if those can be fixed